### ISSUE TITLE
docs: converge on a common set of example images

### ISSUE THREAD

**Issue by thaJeztah** [docker/cli#4902]
### Description

Possibly low-hanging fruit, and perhaps more things to discuss here; we should make a pass at converging on what images to use in our examples, so that users don’t have to pull different ones when trying new examples. Somewhat leaning towards a combination of `ubuntu` and `alpine`;

```bash
git grep ' debian' **/*.md | wc -l
18
git grep ' ubuntu' **/*.md | wc -l
140
git grep ' alpine' **/*.md | wc -l
45
git grep ' busybox' **/*.md | wc -l
147
```

### Why `alpine` over `busybox`?

`busybox` is nice and small, but doesn't have a package manager, so any example using that is effectively "end of the line. that's it. if you want to expand this example to something more useful, you don't have many options.", so `alpine` is a nice in between; slightly larger, but still small. `alpine` is also used as base-image for various variants of official images (e.g. `nginx:alpine`), so things you learn using it can still be applied to those if the user wants to create small images.

### `alpine` vs `ubuntu`

Using `alpine` vs `ubuntu` depends a bit. `alpine` is definitely smaller, so for "one-off" examples is nicer than `ubuntu`. In some cases, `alpine:latest` may be slightly more clear on "for illustration purposes". Lastly; `ubuntu:latest` strictly speaking doesn't match our "best practices" (`:latest` being a "moving" tag). Admitted, perhaps not the most horrible thing to keep things simple, and because `ubuntu:latest` will always be latest _LTS_, so not completely "random", and still reasonably stable).

Many examples could probably work with `alpine`, but there’s something to be said for keeping `ubuntu` as well, as more people are familiar with working with `ubuntu` than `alpine`, but that would depend on the example I guess (i.e., do we expect them to use the example as a starting point for things they’re doing? and in that case, perhaps expect ubuntu(ish)?)

**Comment by dvdksn** [comment]
I agree that we might as well use Alpine in place of Busybox everywhere.

Having a 2nd option in non Ubuntu or Debian alongside Alpine is necessary in some examples for sure. I have a personal bias towards Debian between these two, but that's a weakly held opinion. Although, (I don't have data for this at hand) something leads me to believe that Debian is a more common option as far as actual base images go. For example: Nginx, Node.js, Postgres, and httpd all have alpine/debian versions. There are some images with ubuntu versions but those seem more rare. Of course Ubuntu is probably more likely familiar as a desktop OS. So I guess it depends which alignment we prioritize there.

**Comment by thaJeztah** [comment]
Good points on the `ubuntu` vs `debian` one. Just a short "blurb";

- `debian` is indeed more commonly used as base image for official images
- `debian` tends to be more "pure"; Ubuntu frequently adds their "special sauce" on top with "mixed" results
- :point_up: their push for "snaps" as installer definitely is more problematic here; likely won't work really when used in a container / Dockerfile.
- `debian` has a strong preference for stability (making it often the platform of choice for server / production)
- BUT: that comes at the cost of "older packages", which ..  not always makes it the choice for developers that want "recent" versions ... so bit of a mixed bag there 😞 
- Looking at downloads, Ubuntu is definitely the most popular though (but no idea in what contexts those downloads are used); also not sure if that strictly matters for examples used.
<img width="394" alt="Screenshot 2024-02-29 at 15 02 32" src="https://github.com/docker/cli/assets/1804568/de81cbf3-8275-4de1-a127-c43c9a26f3f7">
